app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Admin::DeployGroupsController < ApplicationController
+  before_action :authorize_admin!
+  before_action :authorize_super_admin!, only: [ :create, :new, :update, :destroy ]
+
+  rescue_from ActiveRecord::RecordNotFound do |error|
+    flash[:error] = "Deploy Group not found."
+    redirect_to admin_deploy_group_path
+  end
+
+  def edit
+    @deploy_group = DeployGroup.find(params[:id])
+  end
+
+  def index
+    @deploy_groups = DeployGroup.all
+  end
+
+  def new
+    @deploy_group = DeployGroup.new
+  end
+
+  def create
+    deploy_group = DeployGroup.create!(deploy_group_params)
+    Rails.logger.info("#{current_user.name_and_email} created the DeployGroup #{deploy_group.name}")
+    redirect_to action: 'index'
+  rescue => ex
```

Comment:
if you go with that it doesn't have to be an instance variable and probably use a version of `create` that doesn't raise